code/controllers/subsystem/ambience.dm

```diff
@@ -59,17 +59,17 @@ SUBSYSTEM_DEF(ambience)
 	new_sound = sound(new_sound, repeat = 0, wait = 0, volume = volume*volume_modifier, channel = CHANNEL_AMBIENCE)
 	SEND_SOUND(M, new_sound)
 
-	return rand(min_ambience_cooldown, max_ambience_cooldown)
+	/// gets the length of the sound about to play in SECONDS
+	var/sound_length = new_sound.len
```

Comment:
you got caught by the meme. sound.len is always zero, unless it was returned from a soundquery()